src/shared/cryptsetup-util.h

```diff
@@ -33,4 +33,6 @@ DEFINE_TRIVIAL_CLEANUP_FUNC(struct crypt_device *, sym_crypt_free);
 
 void cryptsetup_enable_logging(struct crypt_device *cd);
 
+#define LUKS2_TOKENS_MAX 32
```

Comment:
Is this arbitrary or is there an upper limit in the kernel/spec?

Reply:
It's an internal limited hardcoded in libcryptsetup right now. Ideally we'd have an API for this. But there currently is none, afaics.

I asked for one now:

https://gitlab.com/cryptsetup/cryptsetup/-/issues/615

Reply:
(I'll update the PR to add a comment about where this comes from)

Reply:
cool, thanks - I'm going through the PR commit by commit, hope to finish tomorrow